File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        int ans = 0, cur=0;
        stack<char>st;
        for(int i=0;i<s.size();i++){
            if(s[i] == '('){
                st.push('(');
                ++cur;
                ans = max(ans, cur);
            }else if(s[i] == ')'){
                st.pop();
                --cur;
            }else{
                continue;
            }
        }
        return max(ans, cur);
    }
};

File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.java
class Solution {
    public int maxDepth(String s) {
        int res = 0, brackets = 0;

        for (char i : s.toCharArray()) {
            if (i == '(') {
                ++brackets;
            } else if (i == ')') {
                --brackets;
            }
            
            res = Math.max(res, brackets);
        }

        return res;
    }
}